Convert EuiTableOfRecords to simpler table component in scripted fields table.

cjcenizal opened the pull request:
Switching to `EuiBasicTable` results in a 12% reduction in code... not bad.
Aaaaand switching to `EuiInMemoryTable` reduced the code by 48%.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/210/)

chrisronline commented on src/core_plugins/kibana/public/management/sections/indices/edit_index_pattern/scripted_fields_table/scripted_fields_table.js:
```diff
@@ -61,34 +59,23 @@ export class ScriptedFieldsTable extends Component {
-      this.setState(this.computeTableState(this.state.criteria, nextProps));
+    if (fieldFilter) {
+      const normalizedFieldFilter = this.props.fieldFilter.toLowerCase();
+      return fields.filter(field => field.name.toLowerCase().includes(normalizedFieldFilter));
```
We probably also need to leave in the `componentWillReceiveProps` and call `getFilteredItems` when either filter changes. We probably need to also refactor `getFilteredItems` to accept the filters or default to getting them off the prop list. We can't always do the latter because we're calling it from `componentWillReceiveProps` and `this.props` will not equal `nextProps` in that function.

See https://github.com/elastic/kibana/pull/16649/commits/119644584f551e0c6899ba553c3f7e262eb2f8d9#diff-3cebde7c25eb5abea1104b7f91a9f8bdR46

chrisronline replied:
After talking this through with @cjcenizal, I think I was promoting a poor pattern of storing computed data which makes it hard to manage the original data set as well as the computed set. It doesn't seem like a good idea to store them _both_ so I think we should move to a pattern where we only store the original data set and calculate the computed data during the `render()` method. To help on performance of potentially recalculating computed data unnecessarily, let's use [reselect](https://github.com/reactjs/reselect) which is a _tiny_ library designed to only call a function when the parameters have changed.

See https://github.com/elastic/kibana/pull/16649/commits/56e76d2c49353e1db1c914b4a7e5663142ec0907

cjcenizal commented:
@chrisronline I addressed your feedback, but I didn't use your reselect suggestion. Can you take another look? Are you OK with implementing reselect in another PR, just so I can get some time to familiarize myself with it?

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/427/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/429/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/474/)
